collect_app/src/main/java/org/odk/collect/android/instancemanagement/autosend/InstanceAutoSendFetcher.kt

```diff
@@ -1,23 +1,32 @@
 package org.odk.collect.android.instancemanagement.autosend
 
+import android.app.Application
 import org.odk.collect.forms.FormsRepository
 import org.odk.collect.forms.instances.Instance
 import org.odk.collect.forms.instances.InstancesRepository
+import org.odk.collect.settings.SettingsProvider
+import org.odk.collect.settings.enums.AutoSend
+import org.odk.collect.settings.enums.StringIdEnumUtils.getAutoSend
 
 object InstanceAutoSendFetcher {
 
     fun getInstancesToAutoSend(
+        application: Application,
         instancesRepository: InstancesRepository,
-        formsRepository: FormsRepository
+        formsRepository: FormsRepository,
+        settingsProvider: SettingsProvider
     ): List<Instance> {
         val allFinalizedForms = instancesRepository.getAllByStatus(
             Instance.STATUS_COMPLETE,
             Instance.STATUS_SUBMISSION_FAILED
         )
 
+        val autoSendSetting =
+            settingsProvider.getUnprotectedSettings().getAutoSend(application)
+
         return allFinalizedForms.filter {
             formsRepository.getLatestByFormIdAndVersion(it.formId, it.formVersion)?.let { form ->
-                form.shouldFormBeSentAutomatically(true)
+                form.shouldFormBeSentAutomatically(autoSendSetting != AutoSend.OFF)
```

Comment:
Is it ok that here we simply check if it's not off but in other places where `shouldFormBeSentAutomatically` is called we check the network state as well by calling `AutoSendSettingsProvider.isAutoSendEnabledInSettings`?
I guess it is because of https://github.com/getodk/collect/pull/6071/files#diff-ccbed5ebca33fb62147ab3c2b6c1d038bedcb5642afd683a911a94e8f01981f7R72 but just wanted to double-check.

Reply:
Yeah my goal was to move that check to `WorkManger`'s network constraints so we don't have to deal with the network provider in here (and we're not wasting battery running the job when the connection type is wrong).